### PR TITLE
Find bounding boxes of multiple labels

### PR DESCRIPTION
**Pull request by mzient**
#### Why we need this PR?
*Pick one, remove the rest*
- It adds new feature: finding bounding boxes in label tensors; needed for RandomObjetBBox

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     * Added a function which outputs a bounding box for each label value found in the input tensor.
     * Traverse a tensor slice by slice and extend bounding boxes corresponding to the label value found in the tensor
     * Keep a bit mask of labels found in the innermost dimension
     * Collapse dimensions with extent 1
 - Affected modules and functionalities:
     * N/A
 - Key points relevant for the review:
     * N/A
 - Validation and testing:
     * Unit tests
 - Documentation (including examples):
     * Doxygen

**JIRA TASK**: DALI-1817

### CODE REVIEW

**Comment by mzient** [comment]
!build

**Comment by mzient** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

File: dali/kernels/imgproc/structure/label_bbox.h [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    if (in.shape[d] == 1)
+      continue;
+    dim_mapping.push_back(d);
+    simpilfied_shape.shape.push_back(in.shape[d]);
+  }
+
+  if (simpilfied_shape.empty()) {
+    // add artificial dimension, so we can process a scalar as a 1D object
+    simpilfied_shape.shape.push_back(1);
+    dim_mapping.push_back(1);
```
[the remark this answers is not in the document]
Comment (mzient) [reply dali/kernels/imgproc/structure/label_bbox.h:192]: Hmmm... yeah.

[the remark this answers is not in the document]
Comment (mzient) [reply dali/kernels/imgproc/structure/label_bbox.h:192]: Apparently, not tested - I'll add a test.

File: dali/kernels/imgproc/structure/label_bbox.h [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+enable_if_t<std::is_integral<T>::value, T> next(T x) {
+  return x + 1;
+}
+
+template <typename T>
+enable_if_t<std::is_floating_point<T>::value, T> next(T x) {
+  return std::nextafter(x, std::numeric_limits<T>::max());
+}
+
+inline bool hit(span<uint32_t> hits, unsigned idx) {
```
[the remark this answers is not in the document]
Comment (mzient) [reply dali/kernels/imgproc/structure/label_bbox.h:50]: Done.

**Comment by mzient** [comment]
!build

**Review by awolant — APPROVED** [review]

**Pull request merged**